src/averageAges.js

```diff
@@ -55,7 +68,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const notOrphans = people.filter(child => people
+    .some(person => onlyWithSon
+      ? (person.name === child.mother) && (child.sex === 'm')
```

Comment:
```suggestion
      ? (person.name === child.mother && child.sex === 'm')
```

Reply:
mb then `  ? person.name === child.mother && child.sex === 'm'` ?

Reply:
I did this brackets only for better readability